core/src/test/java/io/cucumber/core/feature/GluePathTest.java

```diff
@@ -111,7 +127,44 @@ void absolute_windows_path_form_is_not_valid() {
         Executable testMethod = () -> GluePath.parse("C:\\com\\example\\app");
         IllegalArgumentException actualThrown = assertThrows(IllegalArgumentException.class, testMethod);
         assertThat("Unexpected exception message", actualThrown.getMessage(), is(equalTo(
-            "The glue path must have a classpath scheme C:/com/example/app")));
+                "The glue path must have a classpath scheme C:/com/example/app")));
+    }
+
+    @ParameterizedTest
+    @MethodSource("GluePathAndPatternProvider")
+    void warn_when_glue_as_filesystem_path(String gluePath, Matcher<String> logPattern) {
```

Comment:
Passing a matcher here instead of a string gives us some flexibility. We can now pass any sort of check. This means that we don't need to have an empty string if there is no message.